crates/ast/src/ast/mod.rs

```diff
@@ -85,34 +85,41 @@ impl std::ops::Deref for Arena {
 
 /// A list of doc-comments.
 #[derive(Default)]
-pub struct DocComments<'ast>(pub BoxSlice<'ast, DocComment>);
+pub struct DocComments<'ast> {
+    /// The raw doc comments.
+    pub comments: BoxSlice<'ast, DocComment>,
+    /// The parsed Natspec, if it exists.
+    pub natspec: Option<NatSpec<'ast>>,
```

Comment:
can't we merge this inside of DocComment itself? we want to avoid making DocComments bigger and i think coupling makes more sense